osu.Game/Rulesets/IRulesetInfo.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Game.Database;
+
+#nullable enable
+
+namespace osu.Game.Rulesets
+{
+    /// <summary>
+    /// A representation of a ruleset's metadata.
+    /// </summary>
+    public interface IRulesetInfo : IHasOnlineID
+    {
+        /// <summary>
+        /// The user-exposed name of this ruleset.
+        /// </summary>
+        string Name { get; }
+
+        /// <summary>
+        /// An acronym defined by the ruleset that can be used as a permanent identifier.
+        /// </summary>
+        string ShortName { get; }
+
+        /// <summary>
+        /// A string representation of this ruleset, to be used with reflection to instantiate the ruleset represented by this metadata.
+        /// </summary>
+        string InstantiationInfo { get; }
+
+        public Ruleset? CreateInstance()
+        {
+            var type = Type.GetType(InstantiationInfo);
+
+            if (type == null)
+                return null;
+
+            var ruleset = Activator.CreateInstance(type) as Ruleset;
+
+            // overwrite the pre-populated RulesetInfo with a potentially database attached copy.
+            // TODO: figure if we still want/need this after switching to realm.
+            // ruleset.RulesetInfo = this;
```

Comment:
Shouldn't this be uncommented for the time being, since we haven't switched to realm yet?

Reply:
This method is unused for now (the one in `RulesetInfo` is still there). Can remove the interface level method for the time being if that feels better.